src/tests/functional/shared_test_classes/src/subgraph/matmul_split_decompose.cpp

```diff
@@ -0,0 +1,139 @@
+    }
+
+    auto reshape_const = ov::op::v0::Constant::create(element::i64, {reshape_shape_vec.size()}, reshape_shape_vec.data());
+    auto reshape = std::make_shared<ov::op::v1::Reshape>(reshape_productor, reshape_const, false);
+
+    auto transpose_const = ov::op::v0::Constant::create(element::i64, {5}, {2, 0, 3, 1, 4});
+    auto transpose = std::make_shared<ov::op::v1::Transpose>(reshape, transpose_const);
+
+    auto const_zero = ov::op::v0::Constant::create(element::i64, {}, {0});
+    auto const_one = ov::op::v0::Constant::create(element::i64, {}, {1});
+    auto const_two = ov::op::v0::Constant::create(element::i64, {}, {2});
+
+    std::shared_ptr<ov::Node> gather_0;
+    if (enable_fq) {
+        auto fq0 = std::make_shared<opset10::FakeQuantize>(transpose,
+                                                           opset10::Constant::create(element::f32, Shape{}, {0}),
+                                                           opset10::Constant::create(element::f32, Shape{}, {20}),
+                                                           opset10::Constant::create(element::f32, Shape{}, {0}),
+                                                           opset10::Constant::create(element::f32, Shape{}, {254}),
+                                                           255);
+        gather_0 = std::make_shared<ov::op::v1::Gather>(fq0, const_zero /*indices*/, const_zero /*axis*/);
+    } else {
+        gather_0 = std::make_shared<ov::op::v1::Gather>(transpose, const_zero /*indices*/, const_zero /*axis*/);
+    }
+
+    auto gather_1 = std::make_shared<ov::op::v1::Gather>(transpose, const_one /*indices*/, const_zero /*axis*/);
+    auto gather_2 = std::make_shared<ov::op::v1::Gather>(transpose, const_two /*indices*/, const_zero /*axis*/);
+
+    auto mul_const = ov::op::v0::Constant::create(precision, {1}, {2.0});
+    auto mul2 = std::make_shared<ov::op::v1::Multiply>(gather_1, mul_const);
+
+    std::shared_ptr<ov::Node> mul2_fq = mul2;
+    if (enable_fq) {
+        mul2_fq = std::make_shared<opset10::FakeQuantize>(mul2,
+                                                          opset10::Constant::create(element::f32, Shape{}, {0}),
+                                                          opset10::Constant::create(element::f32, Shape{}, {20}),
+                                                          opset10::Constant::create(element::f32, Shape{}, {0}),
+                                                          opset10::Constant::create(element::f32, Shape{}, {254}),
+                                                          255);
+    }
+
+    auto mm_qk = std::make_shared<ov::op::v0::MatMul>(gather_0, mul2_fq, false, true);
+    auto softmax = std::make_shared<ov::op::v1::Softmax>(mm_qk);
+    auto mm_v = std::make_shared<ov::op::v0::MatMul>(softmax, gather_2, false, false);
+
+    function = std::make_shared<ov::Model>(OutputVector{mm_v}, ParameterVector{param}, "MatMulGatherDecompose");
+    abs_threshold = 1e-2f;
```

Comment:
Why do we change default threshold? Is there any expected accuracy loss which can be introduced by the transformation?

Reply:
Removed. My initial version maybe choose not suitable weigths and some const bring accuracy issue.